### PR TITLE
feat: upgrade eth dependencies [APE-728]

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Upgrades `eth-abi` and `eth-account` to be in a range compatible with `web3.py==6.0.0`

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -25,13 +25,14 @@
         "twine",  # Package upload tool
     ],
     "doc": [
-        "myst-parser>=0.18.1,<0.19",  # Tools for parsing markdown files in the docs
-        "Sphinx>=5.3.0,<6.0",  # Documentation generator
-        "sphinx_rtd_theme>=1.2.0,<2",  # Readthedocs.org theme
+        "myst-parser>=0.17.0,<0.18",  # Tools for parsing markdown files in the docs
+        "sphinx-click>=3.1.0,<4.0",  # For documenting CLI
+        "Sphinx>=4.4.0,<5.0",  # Documentation generator
+        "sphinx_rtd_theme>=1.0.0,<2",  # Readthedocs.org theme
```
Comment (fubuloubu) [inline setup.py:31]: These all have to get downgraded to support this?

File: setup.py [hunk 2]
```diff
@@ -65,11 +66,11 @@
     include_package_data=True,
     install_requires=[
         "dataclassy>=0.8.2,<1",
-        "eth-abi>=2.2.0,<4",
-        "eth-account>0.4.0,<1.0.0",
+        "eth-abi==4.0.0b2",
```
[the remark this answers is not in the document]
Comment (unparalleled-js) [reply setup.py]: I pinged them on Discord, not sure! Web3.py 6.0 is using it though.

File: setup.py [hunk 3]
```diff
@@ -4,20 +4,20 @@
...
         "pytest-xdist",  # multi-process runner
         "pytest-cov",  # Coverage analyzer plugin
-        "hypothesis>=6.2.0,<7.0",  # Strategy-based fuzzer
+        "hypothesis>=6.70.0,<7",  # Strategy-based fuzzer
     ],
     "lint": [
-        "black>=22.12.0",  # auto-formatter and linter
-        "mypy>=0.991",  # Static type analyzer
+        "black>=23.1.0,<24",  # auto-formatter and linter
+        "mypy>=1.1.1,<2",  # Static type analyzer
```
Comment (dtdang) [inline setup.py:14]: Are we finally switching to support the newest version of mypy? I know core and most of our other plugins are still v0.991 and we ran into errors with 1.0 with the Starknet plugin before.

Comment (unparalleled-js) [reply setup.py:14]: I would like to upgrade mypy in core as well as a follow-up task.
But these are only lint deps, so they don't depend on core or each other. You can create separate dev envs for each while waiting for extra-dependencies to catch up.

File: setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (unparalleled-js) [reply setup.py]: it was released, and i updated the pin!

**Pull request merged**